CaT/src/Filter/DisplayFilter.php

```diff
@@ -8,87 +8,171 @@
 * Decides which kind of Filter should be displayed and initialize GUI
 */
 class DisplayFilter {
+	const START_PATH = "0";
+
 	protected $sequence;
 	protected $post_values;
 
-	public function __construct(Filters\Sequence $sequence, array $post_values = array(), $position = "") {
+	public function __construct(Filters\Sequence $sequence, array $post_values = array(), $position = "0") {
 		assert(is_string($position));
+
 		$this->sequence = $sequence;
 		$this->post_values = (!empty($post_values)) ? unserialize($post_values) : $post_values;
-		$this->position = ($position != "") ? unserialize($position) : array();
+		$this->position = $position;
+
+		$this->navi = new Navigator($this->sequence, $this->position);
```

Comment:
(new Navigator($this->sequence)->select($this->position);

wenn select den navigator selbst zurückgibt